tests/unit-tests/test_eval_func.py

```diff
@@ -377,7 +384,95 @@ def test_compute_per_class_roc_auc():
     assert per_class_scores[0] == 0.125
     assert per_class_scores[3] == 0.5
 
+def test_ClassificationMetrics():
+    eval_metric_list = ["accuracy", "hit_at_5", "hit_at_10"]
+    metric = ClassificationMetrics(eval_metric_list, multilabel=False)
+
+    assert "accuracy" in metric.metric_comparator
+    assert "accuracy" in metric.metric_function
+    assert "accuracy" in metric.metric_eval_function
+
+    assert "precision_recall" in metric.metric_comparator
+    assert "precision_recall" in metric.metric_function
+    assert "precision_recall" in metric.metric_eval_function
+
+    assert "roc_auc" in metric.metric_comparator
+    assert "roc_auc" in metric.metric_function
+    assert "roc_auc" in metric.metric_eval_function
+
+    assert "f1_score" in metric.metric_comparator
+    assert "f1_score" in metric.metric_function
+    assert "f1_score" in metric.metric_eval_function
+
+    assert "per_class_f1_score" in metric.metric_comparator
+    assert "per_class_f1_score" in metric.metric_function
+    assert "per_class_f1_score" in metric.metric_eval_function
+
+    assert "per_class_roc_auc" in metric.metric_comparator
+    assert "per_class_roc_auc" in metric.metric_function
+    assert "per_class_roc_auc" in metric.metric_eval_function
+
+    assert "hit_at_5" in metric.metric_comparator
+    assert "hit_at_5" in metric.metric_function
+    assert "hit_at_5" in metric.metric_eval_function
+    assert "hit_at_10" in metric.metric_comparator
+    assert "hit_at_10" in metric.metric_function
+    assert "hit_at_10" in metric.metric_eval_function
+
+    signature = inspect.signature(metric.metric_function["hit_at_5"])
+    assert signature.parameters["k"].default == 5
+    signature = inspect.signature(metric.metric_function["hit_at_10"])
+    assert signature.parameters["k"].default == 10
+
+    metric.assert_supported_metric("accuracy")
+    metric.assert_supported_metric("precision_recall")
+    metric.assert_supported_metric("roc_auc")
+    metric.assert_supported_metric("f1_score")
+    metric.assert_supported_metric("per_class_f1_score")
+    metric.assert_supported_metric("per_class_roc_auc")
+    metric.assert_supported_metric("hit_at_5")
+    metric.assert_supported_metric("hit_at_10")
+
+    pass_assert = False
+    try:
+        metric.assert_supported_metric("hit_at_ten")
+        pass_assert = True
+    except:
+        pass_assert = False
+    assert not pass_assert
+
+def test_compute_hit_at_classification():
+    preds = th.arange(100) / 102
+    # preds = [probe_of_0, probe_of_1]
```

Comment:
Can remove this line?

Reply:
We should keep this to tell how preds looks like.